doc/cookbook/namedRoutes/NamedRoutes.lhs

```diff
@@ -0,0 +1,371 @@
+    |
+226 | server = versionHandler
+```
+
+On the contrary, with the `NamedRoutes` technique, we refer to the routes by their name:
+
+```haskell,ignore
+data API mode = API
+    { list   :: "list" :> ...
+    , delete ::  "delete" :> ...
+    }
+```
+
+and GHC follows the lead :
+
+```console
+    • Couldn't match type 'NoContent' with 'Movie'
+      Expected type: AsServerT Handler :- Delete '[JSON] Movie
+        Actual type: Handler NoContent
+    • In the 'delete' field of a record
+      In the expression:
+        MovieAPI
+          {get = getMovieHandler movieId,
+           update = updateMovieHandler movieId,
+           delete = deleteMovieHandler movieId}
+      In an equation for 'movieHandler':
+          movieHandler movieId
+            = MovieAPI
+                {get = getMovieHandler movieId,
+                 update = updateMovieHandler movieId,
+                 delete = deleteMovieHandler movieId}
+    |
+252 |     , delete = deleteMovieHandler movieId
+```
+
+So, NamedRoutes is more readable for a human, and GHC gives you more accurate error messages.
+
+What are we waiting for?
+
+
+## Boilerplate time!
+
+First, let’s get rid of the the extensions and imports boilerplate in order to focus on our new technique:
+
+
+```haskell
+{-# LANGUAGE DataKinds          #-}
+{-# LANGUAGE DeriveAnyClass     #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DeriveGeneric      #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE TypeOperators      #-}
+
+import GHC.Generics               ( Generic )
+import Data.Aeson                 ( FromJSON, ToJSON )
+import Data.Proxy                 ( Proxy(..) )
+import Network.Wai.Handler.Warp   ( run )
+
+import Servant                    ( NamedRoutes
+                                  , Handler, serve )
+import Servant.API                (Capture, Delete, Get, Put, QueryParam, ReqBody
+                                  , JSON, NoContent (..)
+                                  , FromHttpApiData (..),ToHttpApiData(..)
+                                  , (:>) )
+import Servant.API.Generic        ( (:-) )
+
+import Servant.Client             ( AsClientT, ClientM, client
+                                  , (//), (/:) )
+import Servant.Client.Generic     ()
+
+import Servant.Server             ( Application, ServerT )
+import Servant.Server.Generic     ( AsServerT )
+
+```
+
+## Domain context
+
+Now that we’ve handled the boilerplate, we can dive into our Movie Catalog domain.
+
+Consider a `Movie` constructed from a `Title` and a `Year` of publication.
+
+``` haskell
+data Movie = Movie
+    { movieId :: MovieId
+    , title :: Title
+    , year :: Year
+    }
+    deriving stock Generic
+    deriving anyclass (FromJSON, ToJSON)
+
+type MovieId = String
+type Title = String
+type Year = Int
+
+```
+
+
+Let’s forget about the deriving stuff for now and think about the API that we want to make.
+
+  "version" -> Get Version
```

Comment:
Now that I think of it, we probably want this diagram in a code block (otherwise it won't be formatted properly).